### DISCUSSION TITLE
How do you make Apscheduler run at specific time from sql database

### DISCUSSION THREAD

**Discussion by IfeOduyemi** [agronholm/apscheduler#717] [Q&A]
does anyone know how to turn sql DATETIME type into python datetime type because I want APScheduler to run at a specific time. ``` conn = sqlite3.connect('finance.db')
cursor = conn.cursor()
time_event = cursor.execute("SELECT events_time FROM events WHERE id = 28")
user_time = (" ".join(time_event))
sql_datetime_obj = datetime.datetime.strptime(user_time, "%Y-%m-%d %H:%M:%S")
conn.close()
scheduler = BackgroundScheduler(daemon=True)
scheduler.add_job(send_mail, 'date', run_date=sql_datetime_obj)
scheduler.start()
 ```

**Comment by agronholm** [comment]
What's the problem with that code (other than it exits right after starting the background scheduler)?

**Reply by IfeOduyemi** [reply]
the problem is that there is an attributeerror and its says type object 'datetime.datetime' has no attribute 'datetime'

**Reply by agronholm** [reply]
Then drop the `.datetime` from the `strptime()` call. You're trying to access the `datetime` attribute in the `datetime` class which won't work.

**Reply by IfeOduyemi** [reply]
so sql_datetime_obj = strptime(user_time, "%Y-%m-%d %H:%M:%S")
it now says strptime is not defined do i have to import it?

**Reply by agronholm** [reply]
Now you dropped the entire `datetime.datetime`, when I told you to only strip the `.datetime` part.

**Reply by IfeOduyemi** [reply]
sorry yes i did that but it said that 'time data '' does not match format '%Y-%m-%d %H:%M:%S'

**Reply by agronholm** [reply]
Then it looks like the input data is not a timestamp at all. Check your input.

**Reply by IfeOduyemi** [reply]
yes thank you sorry my input was a text
